components/postgresql/postgresql.app.mjs

```diff
@@ -1,11 +1,215 @@
+      return rows[0];
+    },
+    /**
+     * Gets the number of rows in a table
+     * @param {string} table - Name of database table to query
+     * @returns Count of how many rows exist in the table
+     */
+    async getRowCount(table) {
+      const response = await this.executeQuery(`SELECT COUNT(*) FROM ${table}`);
+      return response[0].count;
+    },
+    /**
+     * Deletes a row or rows in a table based on a lookup column & value
+     * @param {string} table - Name of database table to delete from
+     * @param {string} column - Column used to find the row(s) to delete
+     * @param {string} value - A column value. Used to find the row(s) to delete
+     */
+    async deleteRows(table, column, value) {
+      await this.executeQuery(`DELETE FROM ${table} WHERE ${column} = '${value}'`);
+    },
+    /**
+     * Inserts a row in a table
+     * @param {string} table - Name of database table to insert row into
+     * @param {array} columns - Array of column names
+     * @param {array} values - Array of values corresponding to the column names provided
+     * @returns The newly created row
+     */
+    async insertRow(table, columns, values) {
+      return this.executeQuery(`INSERT INTO ${table} (${columns}) VALUES (${values.map((v) => "'" + v + "'")}) RETURNING *`);
+    },
+    /**
+     * Updates a row in a table
+     * @param {string} table - Name of database table to update
+     * @param {string} lookupColumn - Column used to find row to update
+     * @param {string} lookupValue - A column value. Used to find row to update
+     * @param {object} rowValues - An object with keys representing column names
+     *  and values representing new column values
+     * @returns The newly updated row
+     */
+    async updateRow(table, lookupColumn, lookupValue, rowValues) {
+      const primaryKey = await this.getPrimaryKey(table);
+      let query = `UPDATE ${table} SET`;
+      Object.entries(rowValues).forEach(([
+        key,
+        val,
+      ]) => {
+        query += ` ${key} = '${val}',`;
+      });
+      // trim last comma
+      query = query.slice(0, -1);
+      query += ` WHERE ${primaryKey} =(SELECT ${primaryKey} FROM ${table} WHERE ${lookupColumn} = '${lookupValue}' ORDER BY ${primaryKey} LIMIT 1) RETURNING *`;
+      const response = await this.executeQuery(query);
+      return response[0];
+    },
+    /**
+     * Gets all of the values for a single column in a table
+     * @param {string} table - Name of database table to query
+     * @param {string} column - Name of column to get values from
+     * @params {string} [limit] - maximum number of rows to return
+     * @params {integer} [offset] - number of rows to skip, used for pagination
+     * @returns Array of column values
+     */
+    async getColumnValues(table, column, limit = "ALL", offset = 0) {
+      const rows = await this.executeQuery(`SELECT ${column} FROM ${table} LIMIT ${limit} OFFSET ${offset}`);
```

Comment:
If you change `executeQuery` the way I suggested previously I would refactor this one to
```suggestion
      const rows = await this.executeQuery({
        text: format(`
          SELECT %I FROM %I
            LIMIT $1::numeric OFFSET $2::numeric
        `, column, table),
        values: [
          limit,
          offset,
        ],
      });
```